tests/Database/DatabaseConcernsHasAttributesTest.php

```diff
@@ -61,6 +62,24 @@ public function testUnsettingCachedAttribute()
 
         $this->assertFalse($instance->cachedAttributeIsset('cacheableProperty'));
     }
+
+    #[RequiresPhp('>= 8.4.0')]
+    public function testCanCastAttributeWithGetPropertyHook()
+    {
+        $class = require __DIR__.'/stubs/EloquentModelWithPropertyHook.php';
+        $class->first_name = 'John';
+        $class->last_name = 'Doe';
+        $this->assertSame('John Doe', $class->full_name);
+    }
+
+    #[RequiresPhp('>= 8.4.0')]
+    public function testCanCastAttributeWithSetPropertyHook()
+    {
+        $class = require __DIR__.'/stubs/EloquentModelWithPropertyHook.php';
+        $class->full_name = 'John Doe';
```

Comment:
You need to make sure the following cases also mutate the underlying attributes:
- `new Class(['full_name' => 'John Doe'])`
- `$class->fill(['full_name' => 'John Doe'])`
- `$class['full_name'] = 'John Doe'`